irr/src/CGLTFMeshFileLoader.cpp

```diff
@@ -0,0 +1,693 @@
+	constexpr tiniergltf::Accessor::ComponentType SelfType::Accessor<std::array<T, n>>::getComponentType() \
+	{                                                                                                      \
+		return tiniergltf::Accessor::ComponentType::U;                                                     \
+	}                                                                                                      \
+	template <>                                                                                            \
+	std::array<T, n> SelfType::rawget(const void *ptr)                                                     \
+	{                                                                                                      \
+		const T *tptr = reinterpret_cast<const T *>(ptr);                                                  \
+		std::array<T, n> res;                                                                              \
+		for (u8 i = 0; i < n; ++i)                                                                         \
+			res[i] = rawget<T>(tptr + i);                                                                  \
+		return res;                                                                                        \
+	}
+
+#define ACCESSOR_PRIMITIVE(T, U) \
+	ACCESSOR_TYPES(T, SCALAR, U) \
+	VEC_ACCESSOR_TYPES(T, U, 2)  \
+	VEC_ACCESSOR_TYPES(T, U, 3)  \
+	VEC_ACCESSOR_TYPES(T, U, 4)
+
+ACCESSOR_PRIMITIVE(f32, FLOAT)
+ACCESSOR_PRIMITIVE(u8, UNSIGNED_BYTE)
+ACCESSOR_PRIMITIVE(u16, UNSIGNED_SHORT)
+ACCESSOR_PRIMITIVE(u32, UNSIGNED_INT)
+
+ACCESSOR_TYPES(core::vector3df, VEC3, FLOAT)
+
+template <class T>
+T SelfType::Accessor<T>::get(std::size_t i) const
+{
+	// Buffer-based accessor: Read directly from the buffer.
+	if (std::holds_alternative<BufferSource>(source)) {
+		const auto bufsrc = std::get<BufferSource>(source);
+		return rawget<T>(bufsrc.ptr + i * bufsrc.byteStride);
+	}
+	// Array-based accessor (used for sparse accessors): Read from array.
+	if (std::holds_alternative<std::vector<T>>(source)) {
+		return std::get<std::vector<T>>(source)[i];
+	}
+	// Default-initialized accessor.
+	// We differ slightly from glTF here in that
+	// we default-initialize quaternions and matrices properly,
+	// but this does not cause any discrepancies for valid glTF models.
+	std::get<std::tuple<>>(source);
+	return T();
+}
+
+// Note: clang and gcc should both optimize this out.
+static inline bool isBigEndian()
+{
+	const u16 x = 0xFF00;
+	return *(const u8 *)(&x);
+}
+
+template <typename T>
+T SelfType::rawget(const void *ptr)
+{
+	if (!isBigEndian())
+		return *reinterpret_cast<const T *>(ptr);
+	// glTF uses little endian.
+	// On big-endian systems, we have to swap the byte order.
+	// TODO test this on a big endian system
+	const u8 *bptr = reinterpret_cast<const u8 *>(ptr);
+	u8 bytes[sizeof(T)];
+	for (std::size_t i = 0; i < sizeof(T); ++i) {
+		bytes[sizeof(T) - i - 1] = bptr[i];
+	}
```

Comment:
this function is only for scalar types, right?
in that case better use `os::Byteswap::byteswap`